Kernel/SortHelper.cpp

```diff
@@ -88,10 +88,19 @@ TermList SortHelper::getResultSort(const Term* t)
     return TermList(AtomicSort::superSort());
   }
 
-  Substitution subst;
-  getTypeSub(t, subst);
   Signature::Symbol* sym = env.signature->getFunction(t->functor());
   TermList result = sym->fnType()->result();
+
+  // fast path for constants
+  // also avoids assertion failures below with $let-bound constants,
+  // which have arity 0 but their sort may contain sort variables (!)
+  if(!t->arity()) {
```

Comment:
According to [this](https://ceur-ws.org/Vol-2162/paper-07.pdf) paper, let binders can be non-constant as well. Not saying I have a better idea for now, just that in general we should expect this in other cases as well.

Reply:
Good catch. By some miracle we actually seem to parse and process these beasts correctly (which then triggers the old assertion again). They could even have type arguments. :-O

Proposal: flag let-bound symbols and exempt them from such assertions. This was my original idea but it seemed too heavy-handed for just constants.